src/midend/programTransformation/ompLowering/omp_simd.cpp

```diff
@@ -736,8 +734,15 @@ void OmpSupport::transOmpSimd(SgNode *node) {
     ROSE_ASSERT(body != NULL);
 
     SgForStatement *for_loop = isSgForStatement(body);
-    ROSE_ASSERT(for_loop != NULL);
+    if (for_loop == NULL) {
+        std::vector<SgNode *> loop_list = NodeQuery::querySubTree(body, V_SgForStatement);
+        ROSE_ASSERT(loop_list.size() >= 1);
+        for_loop = isSgForStatement(loop_list.front());
+    }
+    
+    //ROSE_ASSERT(for_loop != NULL);
     SageInterface::forLoopNormalization(for_loop);
+    //std::cout << for_loop->unparseToString() << std::endl;
```

Comment:
Removing all the code commented out is better, but we can do this after merging this pull request.